apps/server/src/modules/task/uc/task.uc.spec.ts

```diff
@@ -1,21 +1,32 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import { UnauthorizedException } from '@nestjs/common';
-
+import { MikroORM } from '@mikro-orm/core';
 import { PaginationQuery } from '@shared/controller';
-import { Course, ICurrentUser, Task, Lesson } from '@shared/domain';
+import { Course, Task, Lesson, User } from '@shared/domain';
+
 import {
 	userFactory,
 	courseFactory,
 	lessonFactory,
 	taskFactory,
 	submissionFactory,
-	createCurrentTestUser,
+	roleFactory,
+	setupEntities,
 } from '@shared/testing';
 import { MongoMemoryDatabaseModule } from '@shared/infra/database';
-import { TaskRepo } from '@shared/repo';
+import { TaskRepo, UserRepo } from '@shared/repo';
+import { TaskUC } from './task.uc';
+import { TaskAuthorizationService, TaskParentPermission, TaskDashBoardPermission } from './task.authorization.service';
 
-import { TaskUC, TaskDashBoardPermission } from './task.uc';
-import { TaskAuthorizationService, TaskParentPermission } from './task.authorization.service';
+let currentUser!: User;
+let currentPermissions!: TaskDashBoardPermission[];
+
+const setupUser = (permissions: TaskDashBoardPermission[]) => {
+	const role = roleFactory.build({ permissions });
+	currentUser = userFactory.buildWithId({ roles: [role] });
+	currentPermissions = permissions;
+	return currentUser;
```

Comment:
Naming: We use the ICurrentUser Interface and the variable for the token based informations. In this case we really have a user from the user entity. Maybe it make sense that we say user an const user = setupUser([..]); instaent of currentUser.

Reply:
i agree. let's change that. would be simpler as well.